A32NX/html_ui/Pages/VLivery/Liveries/Printer/Printer.js

```diff
@@ -0,0 +1,84 @@
+        this.index = parseInt(url.substring(url.length - 1));
+        const buttons = this.querySelector("#buttons");
+        this.leftArrow = this.querySelector("#arrow-left");
+        this.rightArrow = this.querySelector("#arrow-right");
+        if (this.index == 0) {
+            localStorage.setItem("pages", "[]");
+            buttons.style.display = "none";
+        }
+        this.lines = this.querySelector("#lines");
+        const updateLoop = () => {
+            if (!this._isConnected) {
+                return;
+            }
+            this.Update();
+            requestAnimationFrame(updateLoop);
+        };
+        this._isConnected = true;
+        requestAnimationFrame(updateLoop);
+    }
+    disconnectedCallback() {
+    }
+    Update() {
+        const pages = JSON.parse(localStorage.getItem("pages")) || [];
+
+        if (this.index == 0) {
+            const currentPageID = SimVar.GetSimVarValue("L:A32NX_PAGE_ID", "number") - 1;
+            if (currentPageID >= 0 && pages[currentPageID] == null) {
+                const elements = [];
+                for (let i = 0; i < SimVar.GetSimVarValue('L:A32NX_PRINT_LINES', 'number'); i += 1) {
+                    let line = '';
+                    for (let j = 0; j < SimVar.GetSimVarValue(`L:A32NX_PRINT_LINE_LENGTH_${i}`, 'number'); j += 1) {
+                        line += String.fromCharCode(SimVar.GetSimVarValue(`L:A32NX_PRINT_${i}_${j}`, 'number'));
+                    }
+                    elements.push(line);
+                }
+                pages[currentPageID] = elements;
+                localStorage.setItem("pages", JSON.stringify(pages));
+            }
+        }
+        if (pages == null) {
+            return;
+        }
+        let displayedPage = 0;
+        if (this.index == 0) {
+            displayedPage = pages.length - 1;
+        } else {
+            const pagesPrinted = SimVar.GetSimVarValue("L:A32NX_PAGES_PRINTED", "number");
+            const offset = SimVar.GetSimVarValue("L:A32NX_PRINT_PAGE_OFFSET", "number");
+            displayedPage = pagesPrinted - 1 + offset;
+
+            this.leftArrow.style.display = displayedPage > 0 ? "block" : "none";
+            if (displayedPage < 0) {
+                displayedPage = 0;
+                SimVar.SetSimVarValue("L:A32NX_PRINT_PAGE_OFFSET", "number", (pagesPrinted - 1) * -1);
+            }
+
+            this.rightArrow.style.display = displayedPage < (pagesPrinted - 1) ? "block" : "none";
+            if (displayedPage > (pagesPrinted - 1)) {
+                displayedPage = pagesPrinted - 1;
+                SimVar.SetSimVarValue("L:A32NX_PRINT_PAGE_OFFSET", "number", 0);
+            }
+        }
+
+        let newLines = '';
+        for (const value of (pages[displayedPage] || [])) {
+            newLines += `<span class="line">${value}<br/></span>`;
+        }
+        this.lines.innerHTML = newLines;
```

Comment:
This should not run every frame. Can you make it not update the DOM if nothing changed ?